cli/lib/nf3.mjs

```diff
@@ -212,6 +213,7 @@ class Nf3 {
         nonce: this.nonce,
       };
       this.nonce++;
+      this.txsNotConfirmed++;
```

Comment:
Why did we replace `notConfirmed` with `txsNotConfirmed` with the additional increment here?

Reply:
Incrementing `txsNotConfirmed/notConfirmed` in the function's global scope on top of incrementing it inside the `if` condition later means that we will always print that there is an unconfirmed transaction even when there is none. Because we incrementing twice but decrementing only once. 

To dig deeper, if the function successfully executes `if (this.ethereumSigningKey) { }` condition, then we will return before executing `this.txsNotConfirmed--;` in line `240`. So `Number of unconfirmed transactions is ${this.notConfirmed}` is getting printed with wrong value.
 
To explain what the function is meant to do:
`if (this.ethereumSigningKey) { }` is used below to differentiate if the user is using a locked account or an unlocked account. Duncan added this logic to be backwards compatible with David's old wallet code. Today we won't be running into a situation where line 248 with `return this.web3.eth.sendTransaction(tx);` is reached because we don't use unlocked accounts. So Duncan did not implement lines 224 - 244 for an unlocked account too because they don't get executed. 

Can we revert this function's code to the old code please?


or both scenarions - to proposer or to blockchain. This is wrong because `Number of unconfirmed transactions is ${this.notConfirmed}` 
Could we revert this file to older stage and make no changes here?
The logic is correct